### DISCUSSION TITLE
Does ecode use tree sitter, rope science, and crdt?

### DISCUSSION THREAD

**Discussion by arjpar** [SpartanJ/ecode#86] [Q&A] [answered]
Some of those features are found in new modern editors like Xi, Lapce, Helix, etc.

**Comment by SpartanJ** [comment, marked as answer]
Hi!
ecode currently does not use tree-sitters, I'll probably leverage the feature to the LSP. LSPs have a feature called [semantic tokens](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens) (or semantic highlighting) that usually the underlying implementation is a tree-sitter. I'm exploring the possibilities but I'm still not completely sure what's going to be the adequate solution for ecode. tree-sitters are cool but they are not always necessary or the best solution for the problem IMO. ecode currently uses a simpler approach and is more similar to what vscode's [uses](https://github.com/microsoft/vscode-textmate).
CRDTs are for concurrent document editing. ecode does not support that feature at the moment.

**Reply by arjpar** [reply]
Thanks for the quick response! What about rope science? I take it ecode doesn't?

**Reply by SpartanJ** [reply]
rope science would be could to implement at some point but we are not using it at the moment. These cool technologies are all really great but they don't guarantee a great editor, many great editors don't use any of those three. If you ask me I think that for coding the most interesting tech are the tree-sitters, and I'll probably implement them at some point (it shouldn't be that hard), but I'm still iterating in improving many features and sadly I can't advance at the pace I would like because I'm currently the only developer. It would be great if some devs start to collaborate, but I understand that the code base might be challenging since it's huge and in a totally custom engine. I don't lose hope though.

**Reply by arjpar** [reply]
Thanks for the detailed responses!
